restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -1956,18 +1957,19 @@ private static boolean checkParameterType(final Class<?> type,
     return false;
   }
 
-  private static void addResourceMethods(final Class<?> resourceClass,
-                                                   final ResourceModel model)
+  private static void addResourceMethods(final Class<?> resourceClass, final ResourceModel model)
   {
-    // this ignores methods declared in superclasses (e.g. template methods)
-    for (Method method : resourceClass.getDeclaredMethods())
-    {
-      // ignore synthetic, type-erased versions of methods
-      if (method.isSynthetic())
-      {
-        continue;
-      }
+    // this ignores methods declared in superclasses (e.g. template methods) and synthetic type-erased methods.
+    // We sort methods such that batch finders are always processed before finders. This ensures that any validation
+    // of linked batch finder methods from finders works as expected. We don't care about the order in which other
+    // methods are processed.
+    List<Method> methods = Arrays.stream(resourceClass.getDeclaredMethods())
+        .filter(method -> !method.isSynthetic())
```

Comment:
Instead of sorting this, maybe you can do this cross-method validation in `validateResourceModel` method.

Reply:
I thought about that. It makes the code uglier.